Provide simple tool to download all required test assets and put them some place

`kube-test install-assets <location>`

I checked the setup as you have it (vendored integration framework). In my case (as shown below) *everything works as expected™️*.

However I see that it is a bit strange to call a script from within `./vendor` which then puts some artefacts into `./vendor/github.com/kubernetes-sig-testing/frameworks/integration/assets/bin/`. But - for now - that's the place where our framework expects the binaries to be.

As already said, when it comes to integrating the framework into `k/k` we have already some ideas and did some experiments that will handle that - hopefully - a bit nicer.

Anyway, I implemented two things:
- The script tells you, where it downloaded all the things and of which versions those things are
- The script takes a optional argument, with which you can specify a different download location

Also one strange thing I don't understand:
Why did the script come up with `$GOPATH/src/github.com/pwittrock/assets/bin/` as a download location in your case? Did you call the script in `./vendor/github.com/kubernetes-sig-testing/frameworks/integration/scripts/download-binaries.sh`?

Test case on my machine:
```
~/workspace/kubernetes-playground/go/src/github.com/hoegaarden/testing → cat -n main.go
     1  package main
     2
     3  import (
     4          "fmt"
     5
     6          "github.com/kubernetes-sig-testing/frameworks/integration"
     7  )
     8
     9  func main() {
    10          cp := &integration.ControlPlane{}
    11          if err := cp.Start(); err != nil {
    12                  panic(err)
    13          }
    14          fmt.Println("ControlPlane is up")
    15          if err := cp.Stop(); err != nil {
    16                  panic(err)
    17          }
    18          fmt.Println("ControlPlane is down")
    19  }

~/workspace/kubernetes-playground/go/src/github.com/hoegaarden/testing → go run main.go
panic: fork/exec /home/me/workspace/kubernetes-playground/go/src/github.com/hoegaarden/testing/vendor/github.com/kubernetes-sig-testing/frameworks/integration/assets/bin/etcd: no such file or directory

goroutine 1 [running]:
main.main()
        /home/me/workspace/kubernetes-playground/go/src/github.com/hoegaarden/testing/main.go:12 +0x13e
exit status 2

~/workspace/kubernetes-playground/go/src/github.com/hoegaarden/testing → ./vendor/github.com/kubernetes-sig-testing/frameworks/integration/scripts/download-binaries.sh
About to download a couple of binaries. This might take a while...
<< curl output >>
Done!

~/workspace/kubernetes-playground/go/src/github.com/hoegaarden/testing → go run main.go
ControlPlane is up
ControlPlane is down

~/workspace/kubernetes-playground/go/src/github.com/hoegaarden/testing → echo $GOPATH
/home/me/workspace/kubernetes-playground/go

~/workspace/kubernetes-playground/go/src/github.com/hoegaarden/testing →

```

--- a/integration/scripts/download-binaries.sh
+++ b/integration/scripts/download-binaries.sh
@@ -8,6 +8,13 @@ quiet="-s"
   quiet=""
 }
 
+logEnd() {
+  local msg='done.'
+  [ "$1" -eq 0 ] || msg='Error downloading assets'
+  echo "$msg"
+}
+trap 'logEnd $?' EXIT
+
 # Use BASE_URL=https://my/binaries/url ./scripts/download-binaries to download
 # from a different bucket
 : "${BASE_URL:="https://storage.googleapis.com/k8s-c10s-test-binaries"}"
@@ -17,16 +24,25 @@ os="$(uname -s)"
 os_lowercase="$(echo "$os" | tr '[:upper:]' '[:lower:]' )"
 arch="$(uname -m)"
 
+dest_dir="${1:-"${test_framework_dir}/assets/bin"}"
+etcd_dest="${dest_dir}/etcd"
+kubectl_dest="${dest_dir}/kubectl"
+kube_apiserver_dest="${dest_dir}/kube-apiserver"
+
 echo "About to download a couple of binaries. This might take a while..."
-curl $quiet "${BASE_URL}/etcd-${os}-${arch}" --output "${test_framework_dir}/assets/bin/etcd"
-curl $quiet "${BASE_URL}/kube-apiserver-${os}-${arch}" --output "${test_framework_dir}/assets/bin/kube-apiserver"
 
-kubectl_version="$(curl https://storage.googleapis.com/kubernetes-release/release/stable.txt)"
+curl $quiet "${BASE_URL}/etcd-${os}-${arch}" --output "$etcd_dest"
+curl $quiet "${BASE_URL}/kube-apiserver-${os}-${arch}" --output "$kube_apiserver_dest"
+
+kubectl_version="$(curl $quiet https://storage.googleapis.com/kubernetes-release/release/stable.txt)"
 kubectl_url="https://storage.googleapis.com/kubernetes-release/release/${kubectl_version}/bin/${os_lowercase}/amd64/kubectl"
-curl ${quiet} "$kubectl_url" --output "${test_framework_dir}/assets/bin/kubectl"
+curl $quiet "$kubectl_url" --output "$kubectl_dest"
 
-chmod +x "${test_framework_dir}/assets/bin/etcd"
-chmod +x "${test_framework_dir}/assets/bin/kube-apiserver"
-chmod +x "${test_framework_dir}/assets/bin/kubectl"
+chmod +x "$etcd_dest" "$kubectl_dest" "$kube_apiserver_dest"
 
-echo "Done!"
+echo    "# destination:"
+echo    "#   ${dest_dir}"
+echo    "# versions:"
+echo -n "#   etcd:            "; "$etcd_dest" --version | head -n 1
+echo -n "#   kube-apiserver:  "; "$kube_apiserver_dest" --version
+echo -n "#   kubectl:         "; "$kubectl_dest" version --client --short

